src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -137,6 +142,7 @@ private PackageSourceMoniker SelectedSource
 
             Loaded += (_, __) =>
             {
+                ReadAssetsFileAndUpdateDependencyLookup();
```

Comment:
reading assets file and updating dependency lookup should be part of search task which will make sure it doesn't delay the ui load.